test/integration/red_test.go

```diff
@@ -120,13 +117,11 @@ func testREDMetricsForHTTPLibrary(t *testing.T, url string) {
 		require.NoError(t, err)
 		// check duration_count has 3 calls and all the arguments
 		require.Len(t, results, 1)
-		if len(results) > 0 {
-			res := results[0]
-			require.Len(t, res.Value, 2)
-			assert.Equal(t, "3", res.Value[1])
-			addr := net.ParseIP(res.Metric["net_sock_peer_addr"])
-			assert.NotNil(t, addr)
-		}
+		res := results[0]
```

Comment:
This PR also updates the version of the `Eventually`  clause library to stop the execution after a `require` assertion fails (actually, when `t.FailNow()` is invoked).

So this kind of length checks in the tests are not needed anymore.